plugins/pci-mei/fu-mei-common.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-mei-common.h"
+
+const gchar *
+fu_mei_common_family_to_string (FuMeiFamily family)
+{
+	if (family == FU_MEI_FAMILY_SPS)
+		return "SPS";
+	if (family == FU_MEI_FAMILY_TXE)
+		return "TXE";
+	if (family == FU_MEI_FAMILY_ME)
+		return "ME";
+	if (family == FU_MEI_FAMILY_CSME)
+		return "CSME";
+	return "AMT";
+}
+
+static gint
+fu_mei_common_cmp_version (FuMeiVersion *vers1, FuMeiVersion *vers2)
+{
+	guint16 vers1buf[] = {
+		vers1->major,
+		vers1->minor,
+		vers1->hotfix,
+		vers1->buildno,
+	};
+	guint16 vers2buf[] = {
+		vers2->major,
+		vers2->minor,
+		vers2->hotfix,
+		vers2->buildno,
+	};
+	for (guint i = 0; i < 4; i++) {
+		if (vers1buf[i] < vers2buf[i])
+			return -1;
+		if (vers1buf[i] > vers2buf[i])
+			return 1;
+	}
+	return 0;
+}
+
+FuMeiIssue
+fu_mei_common_is_csme_vulnerable (FuMeiVersion *vers)
```

Comment:
To any layman looking at this, it's missing the CVE or INTEL-SA identifiers matching the particular firmware versions.  I think that those are important to include, especially as this is going to rev up with the daemon version as more vulnerabilities are disclosed.

Reply:
Intel don't give us those :( If you know of a list I can copy from, I'm happy to do that.

Reply:
Well I think at least some of them you can probably tell by looking at CVE's listed in https://www.intel.com/content/www/us/en/security-center/advisory/intel-sa-00086.html

Reply:
If that's not workable enough, at least need to mention where to look for updated version numbers with the right hotfixes.

Reply:
I did imagine such a thing would be explained in the _Host Security ID Specification_ on the logic that we can update it async with fwupd releases. I think every `FWUPD_SECURITY_ATTR_ID_*` would have a href link in the spec with a lot more detail.

Reply:
OK..